pkg/cmdlets/commit/commit.go

```diff
@@ -13,26 +13,62 @@ import (
 	"github.com/go-openapi/strfmt"
 )
 
+type commitData struct {
+	Hash    string   `locale:"hash,Commit"`
+	Author  string   `locale:"author,Author"`
+	Date    string   `locale:"date,Date"`
+	Message string   `locale:"message,Commit Message"`
+	Changes []string `locale:"changes,Changes"`
+}
+
 func PrintCommit(out output.Outputer, commit *mono_models.Commit, orgs []gmodel.Organization) error {
-	username, err := usernameForID(commit.Author, orgs)
+	data, err := commitDataFromCommit(commit, orgs)
 	if err != nil {
-		return locale.WrapError(err, "err_commit_print_username", "Could not determine username for commit author")
+		return err
 	}
+	out.Print(data)
 
-	out.Print("")
-	out.Print(locale.Tr("print_commit", commit.CommitID.String()))
-	out.Print(locale.Tr("print_commit_author", username))
-	out.Print(locale.Tr("print_commit_date", time.Time(commit.Added).Format(constants.DateTimeFormatUser)))
-	if commit.Message != "" {
-		out.Print(locale.Tr("print_commit_description", commit.Message))
+	return nil
+}
+
+func PrintCommits(out output.Outputer, commits []*mono_models.Commit, orgs []gmodel.Organization) error {
+	var data []commitData
+	for _, c := range commits {
+		d, err := commitDataFromCommit(c, orgs)
+		if err != nil {
+			return err
+		}
+		data = append(data, d)
 	}
-	out.Print("")
-	out.Print(formatChanges(commit))
+	out.Print(data)
 
 	return nil
 }
 
-func formatChanges(commit *mono_models.Commit) string {
+func commitDataFromCommit(commit *mono_models.Commit, orgs []gmodel.Organization) (commitData, error) {
+	username, err := usernameForID(commit.Author, orgs)
+	if err != nil {
+		return commitData{}, locale.WrapError(err, "err_commit_print_username", "Could not determine username for commit author")
+	}
+
+	return commitData{
+		Hash:    shortHash(commit.CommitID.String()),
+		Author:  username,
+		Date:    time.Time(commit.Added).Format(constants.DateTimeFormatUser),
+		Message: commit.Message,
+		Changes: formatChanges(commit),
+	}, nil
+}
+
+func shortHash(commitID string) string {
+	split := strings.Split(commitID, "-")
+	if len(split) == 0 {
+		return ""
+	}
+	return split[0]
+}
```

Comment:
I had to use a short hash as the formatting can be a little odd when we use the full hash. I will file a bug with details of the output with long values in the commit and message columns

Reply:
Story here: https://www.pivotaltracker.com/story/show/175863427